src/commands/generateRunbook.ts

```diff
@@ -49,15 +55,35 @@ const prepareRunbookFile = async (workspaceFolder: vscode.WorkspaceFolder, promp
 
 export const generateRunbook = (secrets: vscode.SecretStorage) => vscode.window.withProgress({ location: vscode.ProgressLocation.Window }, async progress => {
     // Check docker command exists
+    let dockerExists = false;
     try {
-        const res = spawnSync("docker", ["--version"]);
+        const res = spawnSync("docker", ["version"]);
+
         if (res.error) {
             throw res.error;
         }
-    } catch (e) {
-        return vscode.window.showErrorMessage("Docker not found", { modal: true }, "Install Docker Desktop").then((value) => {
-            if (value === "Install Docker Desktop") {
-                vscode.env.openExternal(vscode.Uri.parse("https://www.docker.com/products/docker-desktop"));
+
+        dockerExists = true;
+
+        if (res.status !== 0) {
+            throw new Error(`Docker command exited with code ${res.status} and output the following error: ${res.error || res.stderr.toString()}`);
+        }
+
+    } catch (e: unknown) {
```

Comment:
Normally, I would say this looks like a job for typed exceptions.  It's hard to see from this code that `dockerExists` is being used as a signal into the catch handler that the exception was thrown because of a process that property exited.

Reply:
I made my own exception type for this